src/commands/trialApp/removeTrialApp.ts

```diff
@@ -13,7 +14,13 @@ export async function removeTrialApp(context: IActionContext, node?: TrialAppTre
         node = await ext.tree.showTreeItemPicker<TrialAppTreeItem>(TrialAppTreeItem.contextValue, context);
     }
 
-    ext.context.globalState.update(TrialAppLoginSession, undefined);
-    delete ext.azureAccountTreeItem.trialAppNode;
-    await ext.tree.refresh();
+    const message: string = `Are you sure you want to remove Trial app "${node.client.fullName}"?`;
+    const remove: MessageItem = { title: 'Remove' };
+    window.showInformationMessage(message, { modal: true }, remove, DialogResponses.cancel).then(async (response: MessageItem): Promise<void> => {
```

Comment:
We only use `.then` instead of `await` when it's not a core part of the command. For example, after deploying, we don't await the prompt to "view output"/"upload app settings"/etc. because it's not required and plenty of people ignore the notification. The reason it matters is because the telemetry doesn't get sent until the command finishes and we really want all deploy telemetry regardless of if they ignore that notification.

In this case, it's a modal dialog (aka they can't ignore it) and it's core to the command itself (aka we're fine waiting until they respond to send telemetry).